src/Symfony/Component/Messenger/Tests/Transport/Doctrine/DoctrineIntegrationTest.php

```diff
@@ -0,0 +1,136 @@
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Tests\Transport\Doctrine;
+
+use Doctrine\DBAL\DriverManager;
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\Messenger\Tests\Fixtures\DummyMessage;
+use Symfony\Component\Messenger\Transport\Doctrine\Connection;
+
+/**
+ * @requires pdo_mysql
+ */
+class DoctrineIntegrationTest extends TestCase
+{
+    private $driverConnection;
+    private $connection;
+
+    protected function setUp()
+    {
+        parent::setUp();
+
+        if (!getenv('MESSENGER_DOCTRINE_DSN')) {
+            $this->markTestSkipped('The "MESSENGER_DOCTRINE_DSN" environment variable is required.');
+        }
+        $dsn = getenv('MESSENGER_DOCTRINE_DSN');
+        $this->driverConnection = DriverManager::getConnection(['url' => $dsn]);
+        $this->connection = new Connection([], $this->driverConnection);
+        // call send to auto-setup the table
+        $this->connection->send('...', []);
+        // ensure the table is clean for tests
+        if (false !== strpos($dsn, 'sqlite')) {
+            $this->driverConnection->exec('DELETE FROM messenger_messages');
+        } else {
+            $this->driverConnection->exec('TRUNCATE messenger_messages');
+        }
+
+        // setup the table
+        $reflection = new \ReflectionClass($this->connection);
+        $setup = $reflection->getMethod('setup');
+        $setup->setAccessible(true);
+        $setup->invoke($this->connection);
+    }
+
+    public function testConnectionSendAndGet()
+    {
+        $this->connection->send('{"message": "Hi"}', ['type' => DummyMessage::class]);
+        $encoded = $this->connection->get();
+        $this->assertEquals('{"message": "Hi"}', $encoded['body']);
+        $this->assertEquals(['type' => DummyMessage::class], $encoded['headers']);
+    }
+
+    public function testSendWithDelay()
+    {
+        $now = \DateTime::createFromFormat('U.u', microtime(true));
+        $this->connection->send('{"message": "Hi i am delayed"}', ['type' => DummyMessage::class], 600000);
+
+        $available_at = $this->driverConnection->createQueryBuilder()
+            ->select('m.available_at')
+            ->from('messenger_messages', 'm')
+            ->where('m.body = :body')
+            ->setParameter(':body', '{"message": "Hi i am delayed"}')
+            ->execute()
+            ->fetchColumn();
+
+        $available_at = new \DateTime($available_at);
+
+        $this->assertGreaterThan($now, $available_at);
+    }
+
+    public function testItRetrieveTheFirstAvailableMessage()
+    {
+        // insert messages
+        // one currently handled
+        $this->driverConnection->insert('messenger_messages', [
+            'body' => '{"message": "Hi handled"}',
+            'headers' => json_encode(['type' => DummyMessage::class]),
+            'queue' => 'default',
+            'created_at' => '2019-03-15 12:00:00',
```

Comment:
Can we use the ISO format here or it needs to be this one? 🤔

Reply:
We can. Maybe I must update the method in `Connection` to user the format `Y-m-d\TH:i:s.u` ?

Reply:
Let's do this then. It's going to be `YYYY-MM-DDTHH:mm:ss.sssZ`, with the `DateTime` as UTC. I suggest to use a function rather than having the `->format(...)` call copy/pasted.